### PR TITLE
feat: new check-in lead role

### LINKED ISSUE #315: [Admin] Need new role for check-in leads to promote waitlisted walk-ins
We will have a select group of check-in leads allowed to accept walk-in applicants who were waitlisted.
- Create a new `CHECKIN_LEAD` role which can perform regular organizer duties on the Admin site
- Allow check-in leads to use the waitlist release endpoint

### PR DESCRIPTION
**Pull request by samderanova**
Resolves #315.

### CODE REVIEW

File: apps/api/src/utils/user_record.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class Role(str, Enum):
     REVIEWER = "reviewer"
     ORGANIZER = "organizer"
     VOLUNTEER = "volunteer"
+    CHECKIN_LEAD = "checkin_lead"
```
Comment (taesungh) [inline apps/api/src/utils/user_record.py:19]: Issue: we'll need to update the authorization on the Admin site to allow the check-in lead to serve as an application manager.

File: apps/api/src/routers/admin.py [hunk 2]
```diff
@@ -213,7 +213,7 @@ async def confirm_attendance() -> None:
 @router.post(
     "/waitlist-release/{uid}",
     # TODO: allow check-in leads to release
```
Comment (taesungh) [inline apps/api/src/routers/admin.py:215]: Chore: TODO comment no longer needed.

File: apps/api/src/utils/user_record.py (hunk 1 above)
Comment (taesungh) [reply apps/api/src/utils/user_record.py:19]: Sorry, and also the applicants summary via `ADMIN_ROLES`

Comment (samderanova) [reply apps/api/src/utils/user_record.py:19]: I added `checkin_lead` directly to `ADMIN_ROLES` in [`46f9c93`](https://github.com/HackAtUCI/irvinehacks-site-2024/pull/323/commits/46f9c936668630a8d071b632c8e449b7bdda9a1e)

**Comment by taesungh** [comment]
Also need it one the API side in admin router

**Review by taesungh — APPROVED** [review]
Should work for now, will figure out how to use on Admin Site.

**Review by waalbert — APPROVED** [review]

**Pull request merged**